src/aiperf/endpoints/huggingface_generate.py

```diff
@@ -0,0 +1,132 @@
+        )
+
+        payload: dict[str, Any] = {"inputs": inputs}
+
+        parameters: dict[str, Any] = {}
+        if turn.max_tokens is not None:
+            parameters["max_new_tokens"] = turn.max_tokens
+
+        if model_endpoint.endpoint.extra:
+            parameters.update(model_endpoint.endpoint.extra)
+
+        payload["parameters"] = parameters
+
+        self.debug(lambda: f"Formatted TGI payload: {payload}")
+        return payload
+
+    def parse_response(
+        self, response: InferenceServerResponse
+    ) -> ParsedResponse | None:
+        """Parse TGI response into ParsedResponse.
+
+        Handles both streaming and non-streaming modes.
+        """
+        if self.model_endpoint.endpoint.streaming:
+            return self._parse_streaming(response)
+        return self._parse_non_streaming(response)
+
+    def _parse_non_streaming(
+        self, response: InferenceServerResponse
+    ) -> ParsedResponse | None:
+        """Handle standard (non-streaming) JSON response."""
+        json_obj = response.get_json()
+        if not json_obj:
+            return None
+
+        if isinstance(json_obj, list) and json_obj:
+            text = json_obj[0].get("generated_text")
+        else:
+            text = json_obj.get("generated_text")
+
+        if not text:
+            self.debug(lambda: f"No generated_text in response: {json_obj}")
+            return None
+
+        data = self.make_text_response_data(text)
+        return ParsedResponse(perf_ns=response.perf_ns, data=data)
+
+    def _parse_streaming(
+        self, response: InferenceServerResponse
+    ) -> ParsedResponse | None:
+        """Handle token stream (SSE or chunked) responses."""
+        chunks = []
+        final_text: str | None = None
+        try:
+            for packet in response.packets:
+                # Each packet is an SSEField(name='data', value='...')
+                if packet.name != "data":
+                    continue
+                try:
+                    json_obj = json.loads(packet.value)
+                    generated_text = json_obj.get("generated_text")
+                    if generated_text:
+                        final_text = generated_text
+                        break
+                    token_text = json_obj.get("token", {}).get("text")
+                    if token_text:
+                        chunks.append(token_text)
+                except Exception:
+                    self.debug(lambda: "JSON parse error in packet: {e}")
+                    continue
+        except Exception:
+            self.debug(lambda: "Error reading stream: {e}")
+
+        if final_text is not None:
+            chunks = [final_text]
+
+        if not chunks:
+            self.debug(lambda: "No chunks collected from stream.")
+            return None
+
+        full_text = "".join(chunks)
+        data = self.make_text_response_data(full_text)
+        return ParsedResponse(perf_ns=response.perf_ns, data=data)
```

Comment:
using `response.get_json()` will handle all of the lower SSE level checking for you. It will extract the json from the non-empty "data" packets already.

Take a look at how `ChatEndpoint.extract_chat_response_data` works for inspiration.